service/src/test/java/uk/nhs/adaptors/gp2gp/common/storage/S3StorageConnectorTest.java

```diff
@@ -1,85 +1,101 @@
+        config.setContainerName(BUCKET_NAME);
 
-        s3StorageConnector = new S3StorageConnector(mockS3Client, config);
+        s3StorageConnector = new S3StorageConnector(s3Client, config);
     }
 
-
     @Test
-    void expectExceptionWhenS3ClientCannotDeliverResponse() {
-        S3StorageConnector storageConnector = new S3StorageConnector(S3Client.builder().region(Region.EU_WEST_2).build(), config);
+    void expectExceptionWhenFileDoesNotExist() {
         Exception exception = assertThrows(StorageConnectorException.class,
-                                           () -> storageConnector.downloadFromStorage("s3File"));
+                                           () -> s3StorageConnector.downloadFromStorage("nonexistent-file.txt"));
 
         assertEquals("Error occurred downloading from S3 Bucket", exception.getMessage());
     }
 
     @Test
-    void downloadFromStorageTest() {
-        var mockResponse = mock(GetObjectResponse.class);
-        var mockInputStream = new ByteArrayInputStream("dummy-content".getBytes());
-        var mockResponseInputStream = new ResponseInputStream<>(mockResponse, mockInputStream);
-        final var request = GetObjectRequest.builder().bucket(config.getContainerName()).key(FILE_NAME).build();
+    void downloadFromStorageTest() throws IOException {
 
-        when(mockS3Client.getObject(request)).thenReturn(mockResponseInputStream);
+        String fileContent = "dummy-content";
+        s3Client.putObject(
+            PutObjectRequest.builder().bucket(BUCKET_NAME).key(FILE_NAME).build(),
+            RequestBody.fromString(fileContent));
 
-        var result = s3StorageConnector.downloadFromStorage(FILE_NAME);
+        ResponseInputStream<GetObjectResponse> response = s3StorageConnector.downloadFromStorage(FILE_NAME);
+        String downloadedContent = new String(response.readAllBytes(), StandardCharsets.UTF_8);
 
-        assertNotNull(result);
-        verify(mockS3Client).getObject(request);
+        assertNotNull(response);
+        assertEquals(fileContent, downloadedContent);
     }
 
     @Test
-    void uploadToStorageTest() {
-        final var expectedRequest = PutObjectRequest.builder().bucket(config.getContainerName()).key(FILE_NAME).build();
+    void uploadToStorageTest() throws IOException {
+        InputStream inputStream = new ByteArrayInputStream("upload-content".getBytes(StandardCharsets.UTF_8));
+        long streamLength = inputStream.available();
+
+        s3StorageConnector.uploadToStorage(inputStream, streamLength, FILE_NAME);
 
-        s3StorageConnector.uploadToStorage(is, STREAM_LENGTH, FILE_NAME);
+        HeadObjectResponse response = s3Client.headObject(HeadObjectRequest.builder().bucket(BUCKET_NAME).key(FILE_NAME).build());
 
-        verify(mockS3Client, times(1)).putObject(eq(expectedRequest), any(RequestBody.class));
+        assertNotNull(response);
+        assertEquals(Optional.of(streamLength).get(), response.contentLength());
```

Comment:
Why not download the object again, and assert it's equal to the "upload-content" from the arrange part of the test?